Source/VMware.PSDesiredStateConfiguration/Classes/DscItems.ps1

```diff
@@ -17,58 +17,87 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 #>
 
 <#
-.Description
-Class that defines a dsc configuration
+.DESCRIPTION
+Base class for DSC items
 #>
-class VmwDscConfiguration {
+class DscItem {
     [ValidateNotNullOrEmpty()]
     [string] $InstanceName
 
+    DscItem($InstanceName) {
+        $this.InstanceName = $InstanceName
+    }
+}
+
+<#
+.DESCRIPTION
+Class that defines a DSC configuration
+#>
+class VmwDscConfiguration : DscItem {
+    [VmwDscNode[]] $Nodes
+
+    VmwDscConfiguration($InstanceName, $Nodes) : base($InstanceName) {
+        $this.Nodes = $Nodes
+    }
+}
+
+<#
+.DESCRIPTION
+Class that defines a regular DSC Node
+#>
+class VmwDscNode : DscItem {
     [VmwDscResource[]] $Resources
 
-    VmwDscConfiguration($InstanceName, $Resources) {
-        $this.InstanceName = $InstanceName
+    VmwDscNode($Connection, $Resources) : base($Connection) {
         $this.Resources = $Resources
     }
 }
 
 <#
-.Description
-Class that defines a dsc resource
+.DESCRIPTION
+Class that defines a vSphere DSC Node
 #>
-class VmwDscResource {
-    [ValidateNotNullOrEmpty()]
-    [string] $InstanceName
+class VmwVsphereDscNode : VmwDscNode {
```

Comment:
Why is VmwVsphereDscNode inheriting from VmwDscNode ? From what I've seen, they both work with Connection and Resources and VmwVsphereDscNode does not extend VmwDscNode in any way. I think that both can inherit DscItem.

Reply:
I am using VmwDscNode as a base class for nodes so that i can group them together and not with other DscItems.